### PR TITLE
Use confined path for posix socket

### LINKED ISSUE #8: Use internal snap path for wpan socket base path
The default path for the socket and lock files are at `/run/openthread-wpan0.*`. The snap has no access to these files, so an interface is defined to grant this access on demand:
https://github.com/canonical/openthread-border-router-snap/blob/db2c1fb4ae615e1c01a3f7045d25dbb001c1ccef/snap/snapcraft.yaml#L34-L38

However, when the wpan interface name is different, a more open interface needs to be used, giving access to the whole `/run` directory on the host:
https://github.com/canonical/openthread-border-router-snap/blob/db2c1fb4ae615e1c01a3f7045d25dbb001c1ccef/snap/snapcraft.yaml#L39-L42

A better workaround is to change the base path. This can be done at compile time, as suggested by @prash813:
```
-DOT_CFLAGS=-DOPENTHREAD_POSIX_CONFIG_DAEMON_SOCKET_BASENAME=\"/tmp/openthread-%s\""
```

### PR DESCRIPTION
**Pull request by farshidtz**
Changes the default posix socket basename from `/run/openthread-%s` to `/run/snap.openthead-border-router/openthread-%s`. Confined snaps have access to `/run/snap.$SNAP_NAME` without the need for granting additional permissions.

Closes #8 

Also, changes the LIBDIR and SYSCONFDIR to write the respective build files in the snapcraft part.

I've adopted the configuration from [here](https://github.com/prash813/otbr-snap/blob/afe629595ce0124c28bcc36f9b7e5ecc16661bfc/snap/snapcraft.yaml#L205).

### CODE REVIEW

**Comment by prash813** [comment]
Yes @farshidtz, it is good approach  to use /run/$SNAP_NAME dir for scoket path,  I do not see any pitfall in it.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -181,9 +166,15 @@ parts:
       cp -vr ./script $CRAFT_PART_INSTALL/bin/
       chmod +x $CRAFT_PART_INSTALL/bin/script/otbr-firewall
 
+      OTBR_OPTIONS="-DCMAKE_INSTALL_LIBDIR=${SNAPCRAFT_PART_INSTALL}/lib/"
+      OTBR_OPTIONS+=" -DCMAKE_INSTALL_SYSCONFDIR=${SNAPCRAFT_PART_INSTALL}/etc/"
+      OTBR_OPTIONS+=" -DOT_CFLAGS=-DOPENTHREAD_POSIX_CONFIG_DAEMON_SOCKET_BASENAME=\"/run/snap.openthread-border-router/openthread-%s\""
+      # OTBR_OPTIONS+=" -DCMAKE_BUILD_TYPE=Debug"
+
```
Comment (prash813) [inline snap/snapcraft.yaml:173]: May be you can replace OT_CFLAGS with the following
-DOT_CFLAGS=-DOPENTHREAD_POSIX_CONFIG_DAEMON_SOCKET_BASENAME=\"/run/snap.$CRAFT_PROJECT_NAME/openthread-%s\"

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -97,15 +87,10 @@ apps:
...
     
 
   ot-ctl:
     command: bin/ot-ctl
-    plugs:
-      - system-run-openthread-wpan0
-      - system-run
   
   pskc:
     command: bin/pskc
```
[the remark this answers is not in the document]
Comment (farshidtz) [reply snap/snapcraft.yaml:96]: It is from 7 July 2022, looks abandoned.

File: README.md [hunk 3]
```diff
@@ -36,10 +36,9 @@ Change using `sudo snap openthread-border-router set key="value"`
 
 Connect interfaces to access desired resources:
 ```bash
-# Allow access to specific system files
+# Allow access to required system files
 sudo snap connect openthread-border-router:system-etc-iproute
```
[the remark this answers is not in the document]
Comment (farshidtz) [reply README.md:40]: I think it depends on whether you are logged in. The same goes for `snap install`.

```
$ snap connect chip-tool:bluez 
error: access denied (try with sudo)
```

**Comment by MonicaisHer** [comment]
Testing the snap built from this PR. After starting the snap, the process of forming a Thread network failed on the web GUI. Here are the errors:
```
Aug 16 09:24:54 ubuntu openthread-border-router.otbr-web[332712]: otbr-web[332712]: [ERR ]-WEB-----: OpenThread daemon is not running.
Aug 16 09:24:54 ubuntu openthread-border-router.otbr-web[332712]: otbr-web[332712]: [ERR ]-WEB-----: Wpan service error: 13
```
Full logs: https://pastebin.canonical.com/p/84JdmXJZRw/

**Comment by farshidtz** [comment]
> Testing the snap built from this PR. After starting the snap, the process of forming a Thread network failed on the web GUI due to "OpenThread daemon is not running" error. Here are the errors:
> 
> ```
> Aug 16 09:24:54 ubuntu openthread-border-router.otbr-web[332712]: otbr-web[332712]: [ERR ]-WEB-----: OpenThread daemon is not running.
> Aug 16 09:24:54 ubuntu openthread-border-router.otbr-web[332712]: otbr-web[332712]: [ERR ]-WEB-----: Wpan service error: 13
> ```
> 

Looks like otbr-agent is running but otbr-web doesn't see it. Perhaps `OPENTHREAD_POSIX_CONFIG_DAEMON_SOCKET_BASENAME` cflag is not passed to the web component during build.

**Comment by farshidtz** [comment]
Eventually, I ended up defining another variable called `WEB_CFLAGS` to set the socket name and pass it as target compile options to the web component. It should be easy to upstream this change.

Alternatively, I could simply replace the default in code: `sed -i 's|/run/openthread-%s|'$SNAP_RUN_OTBR'|' src/web/web-service/ot_client.cpp` but that can't be up streamed!

**Comment by farshidtz** [comment]
> Thank you, code looks good. I wasn't able to test the above step 7 - Pair the Thread lighting device.

I also tested it successfully on a fresh Ubuntu Server 22.04.3 (amd64) installation with avahi-daemon and bluez installed with debian packages.

**Pull request merged**